### DISCUSSION TITLE
Join aliases: am I doing it right? Seem quite overcomplicated

### DISCUSSION THREAD

**Discussion by nappa85** [SeaQL/sea-orm#120] [Q&A] [answered]
Let's have a "simple" query like
```sql
SELECT * FROM table1 INNER JOIN table1 AS table2 ON table1.field = table2.id;
```
How should I represent it using crate's types?

```rust
Query::select()
  .from(Entity)
  .join_as(
    JoinType::LeftJoin,
    Entity,
    Alias::new("table2"),
    SimpleExpr::Column({
        let (a, b) = Column::Field.as_column_ref();
        ColumnRef::TableColumn(a, b)
      }).equals(SimpleExpr::Custom("`table2`.`id`".to_owned())
    ))
```

Am I missing some simplier way?

**Comment by tyt2y3** [comment]
Probably. How about these?

https://www.sea-ql.org/SeaORM/docs/advanced-query/more-join

Depending whether you want to use `RelationDef` to simplify things.

**Comment by nappa85** [comment]
Wait, the core of my issue is about joining aliased tables, while joining standard tables everything works smooth, but adding an alias requires adding a lot of code and rely con custom expression, that on my way of view decreases the security given by an ORM system.
I'm thinking at a possible solution, maybe something like that?
```rust
Column::Field.equals(Expr::tbl(Alias::new("table2"), Column::Id))
```
The problem here is that `equals()` requires `IntoIden`, that `SimpleExpr` doesn't impl for obvious reasons.
So how can I generate an `Iden` for an aliased table without requiring custom expressions?

**Comment by tyt2y3** [comment]
Okay sorry but I am confused. So you want to use an alias as the table name? Can you use a unit struct instead?

```rust
#[derive(Iden)]
pub struct Table2;
```

**Comment by nappa85** [comment]
Maybe I'm not getting it, but this doesn't seems to solve the problem.
How am I going to express join conditions, even using a unit struct?

**Comment by tyt2y3** [comment, marked as answer]
I might not exactly get your scenario, but here is an example from SeaQuery

```rust
#[derive(Iden)]
pub struct MyAlias;

let query = Query::select()
    .from(Char::Table)
    .join_as(
        JoinType::RightJoin,
        Font::Table,
        MyAlias,
        Expr::tbl(Font::Table, Font::Id).equals(MyAlias, Char::FontId)
    )
    .to_owned();
```

**Comment by nappa85** [comment]
Yeah, you're right, I'm making confusion between eq and equals...

**Comment by tyt2y3** [comment]
Yes, to clarify, eq is for comparing values while equals is with another expression.
